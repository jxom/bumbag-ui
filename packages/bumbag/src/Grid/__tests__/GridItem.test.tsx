import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { GridItem } from '../GridItem';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<GridItem>Hello world</GridItem>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<GridItem area="a">Hello world</GridItem>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<GridItem use="p">Hello world</GridItem>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with GridItem props', () => {
      const { result } = renderHook(() => GridItem.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<GridItem>{(GridItemProps) => <div {...GridItemProps}>Hello world</div>}</GridItem>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('GridItem.base should render correctly', () => {
    const { container } = render(<GridItem>hello world</GridItem>, {
      theme: { Grid: { Item: { styles: { base: { backgroundColor: 'red' } } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<GridItem>hello world</GridItem>, {
      theme: { Grid: { Item: { defaultProps: { area: 'test', className: 'test' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
