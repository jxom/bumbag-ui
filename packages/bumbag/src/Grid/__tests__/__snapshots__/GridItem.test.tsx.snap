// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: grid-item;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-GridItem"
>
  Hello world
</p>
`;

exports[`composition hook should return with GridItem props 1`] = `
Object {
  "className": "css-134ef9c css-1pwrqjj css-1yuzjtp fp-Box fp-GridItem",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: grid-item;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-GridItem"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: grid-item;
  grid-area: test;
}

<div
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-GridItem"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: grid-item;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-GridItem"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: grid-item;
  grid-area: a;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-GridItem"
>
  Hello world
</div>
`;

exports[`theming GridItem.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: grid-item;
}

.emotion-0 {
  background-color: #ff0000;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-GridItem"
>
  hello world
</div>
`;
